lib/sparrow/fcm_v1.ex

```diff
@@ -45,7 +45,13 @@ defmodule Sparrow.FCM.V1 do
           push_opts
         ) :: sync_push_result | :ok
   def push(h2_worker_pool, notification, opts \\ []) do
-    case Sparrow.FCM.V1.Notification.verify(notification) do
+    verified_notification =
+      notification
+      |> Sparrow.FCM.V1.Notification.verify()
```

Comment:
This function could trigger chain-verify, so that here this is just a single call.

Reply:
This would also mean, that each module would handle only its own struct directly, not top-level notification.